java/client/src/main/java/glide/api/logging/Logger.java

```diff
@@ -0,0 +1,85 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.logging;
+
+import lombok.Getter;
+import lombok.NonNull;
+
+public class Logger {
+    // TODO: consider lazy loading the glide_rs library
+    static {
+        System.loadLibrary("glide_rs");
+    }
+
+    // Enum ordinal is used, so order of variants must be kept the same
+    @Getter
+    public enum Level {
+        DISABLED(-1),
+        ERROR(0),
+        WARN(1),
+        INFO(2),
+        DEBUG(3),
+        TRACE(4);
+
+        private final int level;
+
+        private Level(int level) {
+            this.level = level;
+        }
+    }
+
+    private static Logger instance;
+    private static Level loggerLevel;
+
+    private Logger(@NonNull Level level, String fileName) {
+        loggerLevel = Level.values()[initInternal(level.getLevel(), fileName)];
+    }
+
+    private Logger(String fileName) {
+        this(Level.DISABLED, fileName);
+    }
+
+    private Logger(@NonNull Level level) {
+        this(level, null);
+    }
+
+    private Logger() {
+        this(Level.DISABLED, null);
+    }
+
+    public static void init(Level level, String fileName) {
+        if (instance == null) {
+            instance = new Logger(level, fileName);
+        }
+    }
+
+    public static void init(String fileName) {
+        init(null, fileName);
+    }
+
+    public static void init() {
+        init(null, null);
+    }
+
+    public static void init(Level level) {
+        init(level, null);
+    }
+
+    public static void log(
+            @NonNull Level level, @NonNull String logIdentifier, @NonNull String message) {
+        if (instance == null) {
+            instance = new Logger(Level.DISABLED, null);
+        }
+        if (!(level.getLevel() <= loggerLevel.getLevel())) {
+            return;
+        }
```

Comment:
This check is done by logger core. Maybe we should skip it.
On another side, for better performance it better to duplicate that.

Reply:
Avoid the FFI call - which is expensive - when unnecessary